app/src/main/java/io/apicurio/registry/mt/TenantMetadataService.java

```diff
@@ -16,75 +16,69 @@
-        //we check if profile is prod, because some tests will fail to start quarkus app.
-        //Those tests checks if multitenancy is supported and abort the test if needed
-        if ("prod".equals(ProfileManager.getActiveProfile()) && mtProperties.isMultitenancyEnabled() && !storage.supportsMultiTenancy()) {
-            throw new DeploymentException("Unsupported configuration, \"registry.enable.multitenancy\" is enabled "
-                    + "but the storage implementation being used (" + storage.storageName() + ") does not support multitenancy");
-        }
-        if (mtProperties.isMultitenancyEnabled() && mtProperties.getTenantManagerUrl().isEmpty()) {
-            throw new DeploymentException("Unsupported configuration, \"registry.enable.multitenancy\" is enabled "
-                    + "but the no \"registry.tenant.manager.url\" is provided");
-        }
-
-        if (mtProperties.isMultitenancyEnabled() && mtProperties.isAuthEnabled() && (mtProperties.getTenantManagerAuthUrl().isEmpty() || mtProperties.getTenantManagerAuthUrl().isEmpty() || mtProperties.getTenantManagerClientId().isEmpty() || mtProperties.getTenantManagerClientSecret().isEmpty())) {
-            throw new DeploymentException("Unsupported configuration, \"registry.enable.multitenancy\" is enabled " + "\"registry.enable.auth\" is enabled "
-                    + "but the no auth properties aren't properly configured");
-        }
-
-        if (mtProperties.isMultitenancyEnabled()) {
-            if (mtProperties.isAuthEnabled()) {
-                this.tenantManagerClient = new TenantManagerClientImpl(mtProperties.getTenantManagerUrl().get(), new Auth(mtProperties.getTenantManagerAuthUrl().get(), mtProperties.getTenantManagerAuthRealm().get(), mtProperties.getTenantManagerClientId().get(), mtProperties.getTenantManagerClientSecret().get()));
-            } else {
-                this.tenantManagerClient = new TenantManagerClientImpl(mtProperties.getTenantManagerUrl().get());
-            }
-        }
-    }
+    Maybe<TenantManagerClient> tenantManagerClient;
 
     //TODO create a TenantConfiguration object and only allow the access to it via the tenant context
     //TODO load the TenantConfiguration into the tenant context in the TenantIdResolver(maybe rename that class)
     //TODO cache the TenantConfiguration in TenantIdResolver
     public RegistryTenant getTenant(String tenantId) throws TenantNotFoundException {
-        if (!mtProperties.isMultitenancyEnabled()) {
+        if (tenantManagerClient.isEmpty()) {
             throw new UnsupportedOperationException("Multitenancy is not enabled");
         }
-
         try {
-            return tenantManagerClient.getTenant(tenantId);
+            return tenantManagerClient.get().getTenant(tenantId);
         } catch (RegistryTenantNotFoundException e) {
             throw new TenantNotFoundException(e.getMessage());
         }
     }
 
+    public List<RegistryTenant> getTenantsForDeletion() {
```

Comment:
making this method return the full list tenants to be deleted completely defeats the purpose of having pagination in the tenant-manager api, altough for now this won't cause any issue because we don't expect to have too much data, at least I would suggest to add a comment here with a TODO to note we could improve this in the future"

Reply:
You're right, I'll process the tenants in pages.